internal/core/src/index/IndexFactory.cpp

```diff
@@ -78,6 +79,202 @@ IndexFactory::CreatePrimitiveScalarIndex<std::string>(
+                                                      index_version,
+                                                      index_size_gb,
+                                                      config);
+            has_raw_data =
+                knowhere::IndexStaticFaced<knowhere::fp32>::HasRawData(
+                    index_type, index_version, config);
+            break;
+        default:
+            PanicInfo(
+                milvus::DataTypeInvalid,
+                fmt::format(
+                    "invalid data type to estimate index load resource: {}",
+                    field_type));
+    }
+
+    LoadResourceRequest request{};
+
+    request.has_raw_data = has_raw_data;
+    request.final_disk_cost = resource.value().diskCost;
+    request.final_memory_cost = resource.value().memoryCost;
+    if (knowhere::UseDiskLoad(index_type, index_version) || mmaped) {
+        request.max_disk_cost = resource.value().diskCost;
+        request.max_memory_cost =
+            std::max(resource.value().memoryCost, download_buffer_size_gb);
+    } else {
+        request.max_disk_cost = 0;
+        request.max_memory_cost = 2 * resource.value().memoryCost;
+    }
+    return request;
+}
+
+LoadResourceRequest
+IndexFactory::ScalarIndexLoadResource(
+    DataType field_type,
+    IndexVersion index_version,
+    float index_size,
+    std::map<std::string, std::string>& index_params,
+    bool mmap_enable) {
+    auto config = milvus::index::ParseConfigFromIndexParams(index_params);
+
+    AssertInfo(index_params.find("index_type") != index_params.end(),
+               "index type is empty");
+    std::string index_type = index_params.at("index_type");
+
+    knowhere::expected<knowhere::Resource> resource;
+    float index_size_gb = index_size * 1.0 / 1024.0 / 1024.0 / 1024.0;
+
+    LoadResourceRequest request{};
+    request.has_raw_data = false;
+
+    if (index_type == milvus::index::ASCENDING_SORT) {
+        request.final_memory_cost = index_size_gb;
+        request.final_disk_cost = 0;
+        request.max_memory_cost = 2 * index_size_gb;
+        request.max_disk_cost = 0;
+        request.has_raw_data = true;
+    } else if (index_type == milvus::index::MARISA_TRIE ||
+               index_type == milvus::index::MARISA_TRIE_UPPER) {
+        if (mmap_enable) {
+            request.final_memory_cost = 0;
+            request.final_disk_cost = index_size_gb;
+            request.max_memory_cost = index_size_gb;
+            request.max_disk_cost = index_size_gb;
+        } else {
+            request.final_memory_cost = index_size_gb;
+            request.final_disk_cost = 0;
+            request.max_memory_cost = 2 * index_size_gb;
+            request.max_disk_cost = 0;
+        }
+        request.has_raw_data = true;
+    } else if (index_type == milvus::index::INVERTED_INDEX_TYPE) {
+        request.final_memory_cost = 0;
+        request.final_disk_cost = index_size_gb;
+        request.max_memory_cost = index_size_gb;
```

Comment:
is there any extra memory cost for inverted index ?

Reply:
mmap enabled works for inverted index as well. Since it might mmap to /tmp dir and it still takes one time memory